pages/index.tsx

```diff
@@ -31,53 +35,82 @@ export default function Home() {
   useEffect(() => {
     console.log('effect');
     async function fetchGifts() {
-      const gifts = await getAllGifts();
-      setGiftData(gifts);
+      try {
+        const gifts = await getAllGifts();
+        setGiftData(gifts);
+      } catch (e) {
+        errorFound(e);
+      }
     }
-    fetchGifts().catch((e) => {
-      console.error(e);
-    });
+    void fetchGifts();
   }, []);
 
   async function handleSubmit(e: FormEvent<HTMLElement>) {
-    e.preventDefault();
-    setGiftNameError(false);
-    setReceiverError(false);
-    // this variable is used for checking both inputs
-    // could use return statement instead of errorFound, but it would not give an error message to all invalid inputs. Only the first invalid input.
-    let errorFound = false;
+    try {
+      e.preventDefault();
+      setGiftNameError(false);
+      setReceiverError(false);
+      // this variable is used for checking both inputs
+      // could use return statement instead of errorFound, but it would not give an error message to all invalid inputs. Only the first invalid input.
+      let errorFound = false;
 
-    if (typeof newGiftName !== 'string' || newGiftName.length === 0) {
-      setGiftNameError(true);
-      errorFound = true;
-    }
-    if (typeof newReceiver !== 'string' || newReceiver.length === 0) {
-      setReceiverError(true);
-      errorFound = true;
-    }
-    if (errorFound) {
-      return;
-    }
+      if (typeof newGiftName !== 'string' || newGiftName.length === 0) {
+        setGiftNameError(true);
+        errorFound = true;
+      }
+      if (typeof newReceiver !== 'string' || newReceiver.length === 0) {
+        setReceiverError(true);
+        errorFound = true;
+      }
+      if (errorFound) {
+        return;
+      }
 
-    const generatedUUID = crypto.randomUUID();
-    const newGift: Gift = {
-      name: newReceiver,
-      gift: newGiftName,
-      id: generatedUUID,
-      createdDate: new Date().getTime(),
-    };
+      const newGift: CreateGift = {
+        name: newReceiver,
+        gift: newGiftName,
+        createdDate: new Date().getTime(),
+      };
 
-    const currentGiftList = await getAllGifts();
-    const updatedGiftList = currentGiftList.concat(newGift);
+      const currentGiftList = await getAllGifts();
+      const createdGift = await createGift(newGift);
+      const updatedGiftList = currentGiftList.concat(createdGift);
```

Comment:
Vielä yhden asian huomasin: Tässä turhaan haetaan kaikki lahjat, koska sulta löytyy jo statesta se tieto. Eli voi tehdä näin.
```suggestion
      const createdGift = await createGift(newGift);
      const updatedGiftList = giftData.concat(createdGift);
```